crates/celo-revm/src/handler.rs

```diff
@@ -130,17 +158,10 @@ where
         }
         let caller = ctx.tx().caller();
 
-        // Convert costs to fee currency
-        let fee_currency_context = &evm.ctx().chain().fee_currency_context;
-        let base_fee_in_erc20 = fee_currency_context
-            .celo_to_currency(fee_currency, U256::from(basefee))
-            .map_err(|e| ERROR::from_string(e))?;
-        // Convert base_fee_in_erc20 (U256) to u128 for gas price calculations
-        let base_fee_in_erc20_u128: u128 = base_fee_in_erc20
-            .try_into()
-            .expect("Failed to convert base_fee_in_erc20 to u128: value exceeds u128 range");
-        let effective_gas_price = evm.ctx().tx().effective_gas_price(base_fee_in_erc20_u128);
-        let tip_gas_price = effective_gas_price.saturating_sub(base_fee_in_erc20_u128);
+        let (base_fee_in_erc20, max_allowed_gas_cost) =
+            self.cip64_get_base_fee_and_max_allowed_gas_cost(evm, fee_currency, basefee)?;
+        let effective_gas_price = evm.ctx().tx().effective_gas_price(base_fee_in_erc20);
+        let tip_gas_price = effective_gas_price.saturating_sub(base_fee_in_erc20);
```

Comment:
Should we use a `saturating_sub` here? An underflow is an error that we ignore this way.

Reply:
Ok, nice. This made me found a possible bug. As I said, that validation is performed before. But for every type except the cip64.
So I have to add that validation, but in this place it would be safe to use it like this